src/main/java/org/openmrs/maven/plugins/ModuleInstall.java

```diff
@@ -114,20 +116,107 @@ public void installModule(String serverId, String groupId, String artifactId, St
             if (currentProperties != null) serverId = currentProperties.getName();
         }
         File serverPath = helper.getServerPath(serverId);
+        Artifact artifact = checkCurrentDirectoryForOpenmrsCoreVersion(helper, serverPath);
+
+        if(artifact != null){
+            deployOpenmrsWar(serverPath, artifact);
+        } else {
+            deployModule(groupId, artifactId, version, serverPath);
+        }
+    }
+
+    /**
+     * Deploy openmrs.war file to server
+     * @param serverPath
+     * @param artifact
+     * @return tru if success
+     * @throws MojoExecutionException
+     */
+    private boolean deployOpenmrsWar(File serverPath, Artifact artifact) throws MojoExecutionException {
+        ServerConfig properties = ServerConfig.loadServerConfig(serverPath);
+        List<Element> artifactItems = new ArrayList<Element>();
+        artifactItems.add(artifact.toElement(serverPath.toString()));
+
+        executeMojoPlugin(artifactItems);
+
+        File openmrsCorePath = new File(serverPath, "openmrs-" + properties.getOpenmrsCoreVersion() + ".war");
+        openmrsCorePath.delete();
+
+        properties.setParam("openmrs.platform.version", mavenProject.getVersion());
+        properties.save();
+        return true;
```

Comment:
If you always return true then there is not point to return anything :) Please change to void.